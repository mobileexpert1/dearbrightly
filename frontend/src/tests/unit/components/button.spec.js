import React from 'react';
import { shallow } from 'enzyme';

import Button from '../../../sections/common/Button/Button';

const blackButtonProps = {
    type: 'black',
    text: 'Create',
    onClick: () => {},
};

const clearButtonProps = {
    type: 'clear',
    text: 'Create',
    onClick: () => {},
};

const buttonProps = {
    text: 'Create',
    onClick: () => {},
};

describe('Button', () => {
    it('should render black button correctly', () => {
        const component = shallow(<Button {...blackButtonProps} />);
        expect(component).toMatchSnapshot();
    });

    it('should render clear button correctly', () => {
        const component = shallow(<Button {...clearButtonProps} />);
        expect(component).toMatchSnapshot();
    });

    it('should render default button correctly', () => {
        const component = shallow(<Button {...buttonProps} />);
        expect(component).toMatchSnapshot();
    });
});
