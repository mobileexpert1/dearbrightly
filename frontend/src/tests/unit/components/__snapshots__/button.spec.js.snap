// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render black button correctly 1`] = `
<div
  className="black-button"
  onClick={[Function]}
>
  Create
</div>
`;

exports[`Button should render clear button correctly 1`] = `
<div
  className="clear-button"
  onClick={[Function]}
>
  Create
</div>
`;

exports[`Button should render default button correctly 1`] = `
<div
  className="black-button"
  onClick={[Function]}
>
  Create
</div>
`;
